lib/osf-components/addon/components/subjects/manager/component.ts

```diff
@@ -117,8 +118,15 @@ export default class SubjectManagerComponent extends Component {
         const { selectedSubjects } = this;
 
         try {
-            yield this.model.updateM2MRelationship('subjects', selectedSubjects);
-            yield this.model.hasMany('subjects').reload();
+            const updateResult: ResourceCollectionDocument =
+                yield this.model.updateM2MRelationship('subjects', selectedSubjects);
+            const updatedSubjects = updateResult.data.map(
+                datum => this.store.peekRecord(
+                    'subject',
+                    datum.id,
+                ),
+            );
+            this.model.set('subjects', updatedSubjects);
```

Comment:
After updating the subjects relationship, we take the response and find those subjects in the store and set the relationship on the model manually so that the validation function has most up-to-date info.